lib/chat_api_web/controllers/upload_controller.ex

```diff
@@ -0,0 +1,68 @@
+defmodule ChatApiWeb.UploadController do
+  use ChatApiWeb, :controller
+
+  alias ChatApi.Uploads
+  alias ChatApi.Uploads.Upload
+
+  action_fallback ChatApiWeb.FallbackController
+
+  def create(conn, %{
+        "upload" => upload_params,
+        "message_id" => message_id
+      }) do
+    file_uuid = UUID.uuid4(:hex)
+    filename = String.replace(upload_params.filename, " ", "-")
+
+    unique_filename = "#{file_uuid}-#{filename}"
+    {:ok, image_binary} = File.read(upload_params.path)
+
+    bucket_name = System.get_env("BUCKET_NAME")
+
+    if bucket_name == nil do
+      raise "s3 bucket is not specified"
+    end
+
+    upload_result =
+      ExAws.S3.put_object(bucket_name, unique_filename, image_binary)
```

Comment:
What if we manage to create an isolated module for AWS insertions/deletions/etc...?

Then we would be able to avoid System calls, UUID and File readings through the controller action, it could be like a public API for controllers, like:

```
def create(conn, params) do
  with {:ok, file} <- S3.insert_file(params),
        {:ok, file} <- Uploads.create_file(params) do
    conn
    |> put_status(:created)
    |> render(...)
  end
end
```
(It's just an illustration, you may already have a better idea of how to accomplish this.)